Documentation/ceph-openshift.md

```diff
@@ -3,86 +3,44 @@ title: OpenShift
+- `operator.yaml`: Creates the security context constraints and starts the operator deployment
+- `object.yaml`: Creates an object store with rgw listening on a valid port number for OpenShift
 
 ## TL;DR
 
 To create an OpenShift cluster, the commands basically include:
 
 ```console
 oc create -f crds.yaml -f common.yaml
-oc create -f operator-openshift.yaml
+oc create -f operator.yaml
 oc create -f cluster.yaml
 ```
 
 ## Rook Privileges
 
 To orchestrate the storage platform, Rook requires the following access in the cluster:
 
-* Create `hostPath` volumes, for persistence by the Ceph mon and osd pods
-* Run pods in `privileged` mode, for access to `/dev` and `hostPath` volumes
-* Host networking for the Rook agent and clusters that require host networking
-* Ceph OSDs require host PIDs for communication on the same node
+- Create `hostPath` volumes, for persistence by the Ceph mon and osd pods
+- Run pods in `privileged` mode, for access to `/dev` and `hostPath` volumes
+- Host networking for the Rook agent and clusters that require host networking
+- Ceph OSDs require host PIDs for communication on the same node
 
 ## Security Context Constraints
 
-Before starting the Rook operator or cluster, create the security context constraints needed by the Rook pods. The following yaml is found in `operator-openshift.yaml` under `/cluster/examples/kubernetes/ceph`.
-
-> **NOTE**: Older versions of OpenShift may require `apiVersion: v1`.
+Before starting the Rook operator or cluster, create the security context constraints needed by the
+Rook pods. The files `scc.yaml` and `scc-csi.yaml` need to be injected. They can be found under `/cluster/examples/openshift/ceph`.
 
-```yaml
-kind: SecurityContextConstraints
-apiVersion: security.openshift.io/v1
-metadata:
-  name: rook-ceph
-allowPrivilegedContainer: true
-allowHostNetwork: true
-allowHostDirVolumePlugin: true
-priority:
-allowedCapabilities: []
-allowHostPorts: false
-allowHostPID: true
-allowHostIPC: false
-readOnlyRootFilesystem: false
-requiredDropCapabilities: []
-defaultAddCapabilities: []
-runAsUser:
-  type: RunAsAny
-seLinuxContext:
-  type: MustRunAs
-fsGroup:
-  type: MustRunAs
-supplementalGroups:
-  type: RunAsAny
-allowedFlexVolumes:
-  - driver: "ceph.rook.io/rook"
-  - driver: "ceph.rook.io/rook-ceph"
-volumes:
-  - configMap
-  - downwardAPI
-  - emptyDir
-  - flexVolume
-  - hostPath
-  - persistentVolumeClaim
-  - projected
-  - secret
-users:
-  # A user needs to be added for each rook service account.
-  # This assumes running in the default sample "rook-ceph" namespace.
-  # If other namespaces or service accounts are configured, they need to be updated here.
-  - system:serviceaccount:rook-ceph:rook-ceph-system
-  - system:serviceaccount:rook-ceph:default
-  - system:serviceaccount:rook-ceph:rook-ceph-mgr
-  - system:serviceaccount:rook-ceph:rook-ceph-osd
+```console
+oc create -f scc.yaml scc-csi.yaml
```

Comment:
If we keep the sccs in the same yaml as the operator, no need for separate commands. It's nice that there are no extra commands currently to install on openshift, you just have to use operator-openshift.yaml. Why split them out? I'd rather just delete the scc.yaml example and keep it with the operator.

Reply:
Yes it's nice, but if we do this I cannot import that unmarshal into a go type the SCC.